File: Data Structures & Algorithms/validate-parentheses/submission-13.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []

        for ele in s:
            if ele in "(,{,[":
                st.append(ele)
            else:
                if len(st)==0:
                    return False
                last = st.pop()
                if (ele == ")" and last!="(") or (ele == "}" and last!="{") or (ele == "]" and last!="["):
                    return False
        return len(st)==0
        

File: Data Structures & Algorithms/is-anagram/submission-13.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] ans = new int[26];
        if(s.length()!=t.length()) return false;
        for(int i=0;i<s.length();i++){
            ans[s.charAt(i)-'a']++;
            ans[t.charAt(i)-'a']--;
        }

        for(int ele: ans){
            if(ele!=0){
                return false;
            }
        }
        return true;
    }
}
